### ISSUE TITLE
Consider moving `irods.test.helpers.make_session` into `irods.helpers`

### ISSUE THREAD

**Issue by alanking** [irods/python-irodsclient#590]
It seems that this `irods.test.helpers.make_session` function has grown to be useful beyond just the tests. Should we move it into this module so that it can be used directly? If so, let's create an issue to look into that some other time (i.e. don't do it as part of this change). The existing version in `irods.test.helpers` can then call the new one with its existing default value(s) to minimize the blast radius.

_Originally posted by @alanking in https://github.com/irods/python-irodsclient/pull/589#discussion_r1686700667_

It may or may not be beneficial to do this, but I didn't want to lose the idea, so here's the issue. :)

**Comment by mstfdkmn** [comment]
FYI: I remember this https://github.com/irods/python-irodsclient/issues/412 because waiting for an enhancement to use make_session easily where needed.

**Comment by alanking** [comment]
@mstfdkmn - Good catch. Perhaps this can be closed as a duplicate of #412. Thoughts, @d-w-moore?

**Comment by d-w-moore** [comment]
Yes, this PR probably addresses #412.  As to what's the better name, `irods.utility`, `irods.misc` or `irods.helpers` Have there been any other PR's around that attempt doing the same?

**Comment by korydraughn** [comment]
There's already an `irods.session` module. Why not put it in there?
```python
from irods.session import make_session
```

**Comment by d-w-moore** [comment]
@korydraughn I'm ok with that. So then we keep #412 around for now, and consider whether it may or may not be addressed by the creation of `irods.helpers.xml_mode` and the namespace / package in which it resides?

**Comment by korydraughn** [comment]
Sounds good to me.
